app/src/main/java/ru/alphadrow/gb/myweather/view/main/MainFragment.kt

```diff
@@ -90,12 +93,23 @@ class MainFragment : Fragment(), OnItemViewClickListener {
 
             is AppState.Success -> {
                 binding.mainFragmentLoadingLayout.visibility = View.GONE
-                val weather = appState.weatherData
+                val weather = appState.weatherData.also {
+                    Toast.makeText(context, "ALSO", Toast.LENGTH_SHORT).show()
+                }
                 adapter.setWeather(weather)
+                binding.root.showSnackBarWithAction(R.string.success)
             }
         }
     }
 
+    fun View.showSnackBarWithoutAction(stringId: Int, throwable: Throwable){
+        Snackbar.make(binding.root, getString(stringId), Snackbar.LENGTH_LONG ).show()
+    }
+    fun View.showSnackBarWithAction(stringId: Int){
+        Snackbar.make(binding.root, getString(stringId), Snackbar.LENGTH_LONG).setAction("Action Text",
+            View.OnClickListener { viewModel.getDataFromLocalSource(isDataSetRus) } ).show()
+    }
```

Comment:
```suggestion
    fun View.showSnackBarWithAction(text: String, actionText: String, action: View.OnClickListener) {
    Snackbar.make(this, text, Snackbar.LENGTH_LONG).setAction(actionText, action).show()
}
```
Вот так будет более универсально. 
Если надо именно R.string... ресурсы использовать, то можно `fun View.showSnackBarWithAction(@StringRes text: Int, @StringRes actionText: Int, action: View.OnClickListener)` сделать и дальше получать их через getString, но тогда надо позаботиться о том, что бы showSnackBarWithAction имел context, проще и правильнее в месте вызова из ресурсов получать строку как по мне.